Unused variable warning with Bazel

When compiling with Bazel, I see the following warning:

```
INFO: From Compiling external/jsonnet/core/desugarer.cpp [for host]:
external/jsonnet/core/desugarer.cpp:356:26: warning: unused variable 'index' [-Wunused-variable]
                    AST *index = super_index->index;
                         ^
1 warning generated.
```

The code from that line is https://github.com/google/jsonnet/blob/master/core/desugarer.cpp#L358:

```
AST *index = super_index->index;
// Desugaring of expr should already have occurred.
assert(index != nullptr);

// index is unused after this.
```

An easy fix is to inline the expression but I'm not much of a cpp developer so I don't know what's good style.

--- a/core/desugarer.cpp
+++ b/core/desugarer.cpp
@@ -355,9 +355,8 @@ class Desugarer {
                     UStringStream ss;
                     ss << U"$outer_super_index" << (counter++);
                     const Identifier *super_var = desugarer->id(ss.str());
-                    AST *index = super_index->index;
                     // Desugaring of expr should already have occurred.
-                    assert(index != nullptr);
+                    assert(super_index->index != nullptr);
                     // Re-use super_index since we're replacing it here.
                     superVars.emplace_back(super_var, super_index);
                     expr = alloc.make<Var>(expr->location, expr->openFodder, super_var);

--- a/core/desugarer.cpp
+++ b/core/desugarer.cpp
@@ -404,7 +404,7 @@ class Desugarer {
         return super_vars;
     }
 
-    AST* makeArrayComprehension(ArrayComprehension *ast, unsigned obj_level) {
+    AST* makeArrayComprehension(ArrayComprehension *ast) {
         int n = ast->specs.size();
         AST *zero = make<LiteralNumber>(E, EF, "0.0");
         AST *one = make<LiteralNumber>(E, EF, "1.0");
@@ -697,7 +697,7 @@ class Desugarer {
                 desugar(spec.expr, obj_level);
             desugar(ast->body, obj_level + 1);
 
-            ast_ = makeArrayComprehension(ast, obj_level);
+            ast_ = makeArrayComprehension(ast);
 
         } else if (auto *ast = dynamic_cast<Assert *>(ast_)) {
             desugar(ast->cond, obj_level);

--- a/core/vm.cpp
+++ b/core/vm.cpp
@@ -1435,7 +1435,7 @@ class Interpreter {
             ss << "substr third parameter should be greater than zero, got " << len;
             throw makeError(loc, ss.str());
         }
-        if (from > str->value.size()) {
+        if (static_cast<unsigned long>(from) > str->value.size()) {
             scratch = makeString(UString());
             return nullptr;
         }

